src/metabase/api/card.clj

```diff
@@ -1139,6 +1142,13 @@ saved later when it is ready."
   (when (or is_on_demand is_full_sync)
     (future (sync/sync-table! table))))
 
+(defn csv-stats [csv-file]
+  (with-open [reader (bom/bom-reader csv-file)]
+    (let [rows (csv/read-csv reader)]
```

Comment:
should we makes `load-from-csv` returns these stats so we don't have to read a csv twice?

Reply:
Good point, in the successful case this could be avoided. In the unsuccessful case we need to read the csv because `load-from-csv` could be the function that has a failure. 
Done here [Rename upload_speed to upload_seconds and avoid wasting time if the u…](https://github.com/metabase/metabase/pull/31407/commits/50861c6ad698e6178b9aecda514286b648eaff1b)
Further optimization in the unsuccessful case is possible but imo not worth messing up the code and failures should be rare anyway.

Reply:
is packing those stats info into `ex-data` of the exception too weird?

Reply:
What's the benefit?

Reply:
we don't have to read csv again in unsuccessful cases

Reply:
Ah, I see. I think that fall into the category of "not worth messing up the code".